README.md
# ASPlus - Affiliate Smart Plus

Project to control personal/professional quotes, sales and opportunities.

Develop with Vue and Quasar Framework.

This application is build using [Quasar framework](https://github.com/quasarframework/quasar), based on VueJS.

## Documentation

- [1. Starting](./docs/1.STARTING.md)
- [2. Setup the project](./docs/2.SETUP.md)
- [3. Running the project](./docs/3.HOW_TO_RUN.md)
- [4. Structure of the project](./docs/4.STRUCTURE.md)
- [5. Internationalization i18n](./docs/5.INTERNATIONALIZATION.md)
- [6. Permissions](./docs/6.%20PERMISSIONS.md)
- [7. Building cross-platform apps](./docs/7.BUILD_CROSS_PLATAFORM_APPS.md)

## License

Copyright (c) 2021 - Present by `Dionei Beilke dos Santos`

[MIT License](http://en.wikipedia.org/wiki/MIT_License)


CHANGELOG.md
# [1.0.2]

#### Changed
- Pump project version from `1.0.1` to `1.0.2`
- Enhance the user dashboard (homepage) to get more useless information [#51](https://github.com/BIEMAX/quote-manager/issues/51)
- Change the process of registration 'new presell' from dialog to page
- Always open the menus as opened
- Reduced the notification badge time from '1,5' seconds to '0,7' seconds
- Not allow copy pre-sale page final url if is not present

#### Fix
- Fix tips not loading properly at home screen [#59](https://github.com/BIEMAX/quote-manager/issues/59)
- Fix duplicate success messages after close dialogs [#33](https://github.com/BIEMAX/quote-manager/issues/33)
- Fix warning when select the color for pre-sale pages 'Failed to resolve directive: value at <ColorPicker>'
- Fix shortcut to homepage not working properly [#57](https://github.com/BIEMAX/quote-manager/issues/57)
- Fix data not re-loading properly in tables after do some action [#60](https://github.com/BIEMAX/quote-manager/issues/60)
- Fix not re-loading the grid after pay an overdue bill
- Fix login card not centered at the screen size [#72](https://github.com/BIEMAX/quote-manager/issues/72)

#### Implement
- Get device information on user registration/login [#25](https://github.com/BIEMAX/quote-manager/issues/25)
- Defaulters screen, add and subtract debt and user tracking changes on the values [#35](https://github.com/BIEMAX/quote-manager/issues/35)
- Added vertical scrolls for mobile devices [#42](https://github.com/BIEMAX/quote-manager/issues/42)
- Re-branded the menus and icons [#98](https://github.com/BIEMAX/financial-manager-app/issues/98)
- In pre-sale registration, allow user define the headline [#61](https://github.com/BIEMAX/quote-manager/issues/61)
- In pre-sale creation, added button 'privacy' [#63](https://github.com/BIEMAX/quote-manager/issues/63)
- In pre-sale creation, added button 'terms' [#64](https://github.com/BIEMAX/quote-manager/issues/64)
- In pre-sale creation, added button 'contact' [#65](https://github.com/BIEMAX/quote-manager/issues/65)
- Enable fields based on template selection in pre-sale registration [#70](https://github.com/BIEMAX/quote-manager/issues/70)
- Implemented github actions for automatized deploys for Google Firebase
- In pre-sale page registration, validate if the affiliate url contains malicious content [#67](https://github.com/BIEMAX/quote-manager/issues/67)
- In pre-sale page registration, possibility to change button text [#66](https://github.com/BIEMAX/quote-manager/issues/66)
- In pre-sale page registration, not allow use the product link [#69](https://github.com/BIEMAX/quote-manager/issues/69)
- Added column 'bill type' at the bills grid
- Finish the menu 'pre-sales' implementing edition [#34](https://github.com/BIEMAX/quote-manager/issues/34)

# [1.0.1]

#### Changed
- Change the time notification from '5 milliseconds' to '1,5 milliseconds'
- Change 'user login' to 'user email' [#27](https://github.com/BIEMAX/quote-manager/issues/27)
- Change the the custom hint in 'required inputs' to accept a custom one
- Change the input from 'filled' to 'outlined' [#36](https://github.com/BIEMAX/quote-manager/issues/36)
- Change the selected sub menu to highlight [#37](https://github.com/BIEMAX/quote-manager/issues/37)
- Always start filter panel as opened based on device [#38](https://github.com/BIEMAX/quote-manager/issues/38)
- Removed duplicate translation in i18n config file [#41](https://github.com/BIEMAX/quote-manager/issues/41)
- Enhance the user home screen to get more useless information
- Refactor the user menu permissions based on API information provided
- Changed the pallet colors of the UI [#55](https://github.com/BIEMAX/quote-manager/issues/55)

#### Implement
- Implement sections for 'field to query' and 'buttons' on `Filter panel`
- Build dependency to generate executable for other operational systems [#23](https://github.com/BIEMAX/quote-manager/issues/23)
- Implement 'About section' to check database and backend services [#19](https://github.com/BIEMAX/quote-manager/issues/19)
- Implement 'User history' [#28](https://github.com/BIEMAX/quote-manager/issues/28)
- Implement 'User update info' page [#26](https://github.com/BIEMAX/quote-manager/issues/26)
- Implement menu for 'Google Ads and affiliates' [#24](https://github.com/BIEMAX/quote-manager/issues/24)
- Added function to remove bill from 'finances menu'
- Option to upload pre-sell images [#31](https://github.com/BIEMAX/quote-manager/issues/31)
- Implement function to delete defaulter [#48](https://github.com/BIEMAX/quote-manager/issues/48)
- Implement tags menu [#49](https://github.com/BIEMAX/quote-manager/issues/49)
- Implement new options to register new presell
- Implement design patterns over the buttons for better UX [#56](https://github.com/BIEMAX/quote-manager/issues/56)
- Implement cross platform builder apps through electron [#44](https://github.com/BIEMAX/quote-manager/issues/44)

#### Fix
- Fix `Filter Panel` component not showing properly the components added in template slot
- Fix the buttons in table row not triggering events when clicked [#21](https://github.com/BIEMAX/quote-manager/issues/21)
- Fix invalid date after login [#17](https://github.com/BIEMAX/quote-manager/issues/17)
- Fix 'keep me connected' not working [#22](https://github.com/BIEMAX/quote-manager/issues/22)
- Fix user permissions in submenu options not working properly [#29](https://github.com/BIEMAX/quote-manager/issues/29)
- Fix the menus always expanded after login [#32](https://github.com/BIEMAX/quote-manager/issues/32)
- Fix tables not updating after insert a new register [#43](https://github.com/BIEMAX/quote-manager/issues/43)
- Fix the query data after the dialogs close's [#39](https://github.com/BIEMAX/quote-manager/issues/39)
- Fix a bug that prevents a new bill be registered [#45](https://github.com/BIEMAX/quote-manager/issues/45)
- Fix bill list not updating after dialog close [#47](https://github.com/BIEMAX/quote-manager/issues/47)

quasar.config.js
/* eslint-env node */

/*
 * This file runs in a Node context (it's NOT transpiled by Babel), so use only
 * the ES6 features that are supported by your Node version. https://node.green/
 */

// Configuration for your app
// https://v2.quasar.dev/quasar-cli-webpack/quasar-config-js

/* eslint-disable @typescript-eslint/no-var-requires */

const { configure } = require('quasar/wrappers')
const ESLintPlugin = require('eslint-webpack-plugin')
const path = require('path')

module.exports = configure(function (ctx) {
  return {
    // https://v2.quasar.dev/quasar-cli-webpack/supporting-ts
    supportTS: {
      tsCheckerConfig: {
        eslint: {
          enabled: true,
          files: './src/**/*.{ts,tsx,js,jsx,vue}'
        }
      }
    },
    boot: [
      'i18n',
      'axios'
    ],
    // https://v2.quasar.dev/quasar-cli-webpack/quasar-config-js#Property%3A-css
    css: ['app.scss'],
    // https://github.com/quasarframework/quasar/tree/dev/extras
    extras: [
      'ionicons-v4',
      'mdi-v6',
      'mdi-v7',
      'fontawesome-v6',
      'eva-icons',
      'themify',
      'line-awesome',
      'roboto-font-latin-ext', // this or either 'roboto-font', NEVER both!
      // 'roboto-font', // optional, you are not bound to it
      'material-icons' // optional, you are not bound to it
    ],

    // Full list of options: https://v2.quasar.dev/quasar-cli-webpack/quasar-config-js#Property%3A-build
    build: {
      vueRouterMode: 'hash', // available values: 'hash', 'history'
      // transpile: false,
      // publicPath: '/',

      // Add dependencies for transpiling with Babel (Array of string/regex)
      // (from node_modules, which are by default not transpiled).
      // Applies only if "transpile" is set to true.
      // transpileDependencies: [],

      // rtl: true, // https://quasar.dev/options/rtl-support
      // preloadChunks: true,
      // showProgress: false,
      // gzip: true,
      // analyze: true,

      // Options below are automatically set depending on the env, set them if you want to override
      // extractCSS: false,

      // https://v2.quasar.dev/quasar-cli-webpack/handling-webpack
      // "chain" is a webpack-chain object https://github.com/neutrinojs/webpack-chain
      // chainWebpack (/* chain */) {},

      // devtool: 'source-map',
      distDir: 'dist',
      chainWebpack: chain => {
        chain.module
          .rule('i18n-resource')
          .test(/\.(json5?|ya?ml)$/)
          .include.add(path.resolve(__dirname, './src/i18n'))
          .end()
          .type('javascript/auto')
          .use('i18n-resource')
          .loader('@intlify/vue-i18n-loader')
        chain.module
          .rule('i18n')
          .resourceQuery(/blockType=i18n/)
          .type('javascript/auto')
          .use('i18n')
          .loader('@intlify/vue-i18n-loader')
      }
    },

    // Full list of options: https://v2.quasar.dev/quasar-cli-webpack/quasar-config-js#Property%3A-devServer
    devServer: {
      server: {
        type: 'http'
      },
      port: 8001,
      open: true // opens browser window automatically
    },

    // https://v2.quasar.dev/quasar-cli-webpack/quasar-config-js#Property%3A-framework
    framework: {
      config: {},

      // iconSet: 'material-icons', // Quasar icon set
      lang: 'en-US', // Quasar language pack

      // For special cases outside of where the auto-import strategy can have an impact
      // (like functional components as one of the examples),
      // you can manually specify Quasar components/directives to be available everywhere:
      //
      // components: [],
      // directives: [],

      // Quasar plugins
      plugins: ['Dialog', 'Notify', 'Loading', 'Cookies']
    },

    // animations: 'all', // --- includes all animations
    // https://quasar.dev/options/animations
    animations: [],

    // https://v2.quasar.dev/quasar-cli-webpack/developing-ssr/configuring-ssr
    ssr: {
      pwa: false,

      // manualStoreHydration: true,
      // manualPostHydrationTrigger: true,

      prodPort: 3000, // The default port that the production server should use
      // (gets superseded if process.env.PORT is specified at runtime)

      maxAge: 1000 * 60 * 60 * 24 * 30,
      // Tell browser when a file from the server should expire from cache (in ms)

      // chainWebpackWebserver (/* chain */) {},

      middlewares: [
        ctx.prod ? 'compression' : '',
        'render' // keep this as last one
      ]
    },

    // https://v2.quasar.dev/quasar-cli-webpack/developing-pwa/configuring-pwa
    pwa: {
      workboxPluginMode: 'GenerateSW', // 'GenerateSW' or 'InjectManifest'
      workboxOptions: {}, // only for GenerateSW

      // for the custom service worker ONLY (/src-pwa/custom-service-worker.[js|ts])
      // if using workbox in InjectManifest mode
      // chainWebpackCustomSW (/* chain */) {},

      manifest: {
        name: 'AffiliateSmartPlus',
        short_name: 'ASPlus',
        description: '',
        display: 'standalone',
        orientation: 'portrait',
        background_color: '#ffffff',
        theme_color: '#027be3',
        icons: [
          {
            src: 'icons/icon-128x128.png',
            sizes: '128x128',
            type: 'image/png'
          },
          {
            src: 'icons/icon-192x192.png',
            sizes: '192x192',
            type: 'image/png'
          },
          {
            src: 'icons/icon-256x256.png',
            sizes: '256x256',
            type: 'image/png'
          },
          {
            src: 'icons/icon-384x384.png',
            sizes: '384x384',
            type: 'image/png'
          },
          {
            src: 'icons/icon-512x512.png',
            sizes: '512x512',
            type: 'image/png'
          }
        ]
      }
    },

    // Full list of options: https://v2.quasar.dev/quasar-cli-webpack/developing-cordova-apps/configuring-cordova
    cordova: {
      // noIosLegacyBuildFlag: true, // uncomment only if you know what you are doing
    },

    // Full list of options: https://v2.quasar.dev/quasar-cli-webpack/developing-capacitor-apps/configuring-capacitor
    capacitor: {
      hideSplashscreen: true
    },

    // Full list of options: https://v2.quasar.dev/quasar-cli-webpack/developing-electron-apps/configuring-electron
    electron: {
      bundler: 'packager', // 'packager' or 'builder'

      // packager: {
      //   // https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#options

      //   // OS X / Mac App Store
      //   // appBundleId: '',
      //   // appCategoryType: '',
      //   // osxSign: '',
      //   // protocol: 'myapp://path',

      //   // Windows only
      //   // win32metadata: { ... }
      // },
      packager: {
        extends: null, // Fix error 'The main entry point to your app was not found. Make sure "index.js" exists and does not get ignored by your ignore option'
        platform: ['win32', 'linux', 'darwin'],
        arch: 'x64', // or 'ia32' or 'armv7l' or 'arm64'
        icon: 'src-electron/icons/icon.ico',
        dir: 'dist/electron',
        out: 'build',
        asar: true,
        overwrite: true,
        win32metadata: {
          ProductName: 'AS Plus Manager',
          CompanyName: 'SmartThinq Technologies LTDA.',
          FileDescription: 'App to control finances ',
          OriginalFilename: 'MyApp.exe',
          InternalName: 'AS Plus',
          PrivateBuild: 'Built by Quasar',
          FileVersion: '1.0.1.0',
          ProductVersion: '1.0.1.0'
        }
      },

      chainWebpackMain (chain) {
        chain
          .plugin('eslint-webpack-plugin')
          .use(ESLintPlugin, [{ extensions: ['js'] }])
      },

      chainWebpackPreload (chain) {
        chain
          .plugin('eslint-webpack-plugin')
          .use(ESLintPlugin, [{ extensions: ['js'] }])
      }
    }
  }
})
